src/hooks/useFetchPosts.js

```diff
@@ -39,8 +39,51 @@ const fetchPosts = async (subreddit, posts = [], after = null, maxPosts = 500) =
   return fetchPosts(subreddit, newPosts, currentAfter);
 };
 
+
+const transformPostData = (post) => ({
+  author: post.data.author,
+  createdAt: post.data.created_utc,
+  name: post.data.name,
+  numComments: post.data.num_comments,
+  permalink: post.data.permalink,
+  score: post.data.score,
+  title: post.data.title,
+});
+
+const getPostsAndCountByHour = (posts) => {
+  const countsByHour = Array(7)
+    .fill()
+    .map(() => Array(24).fill(0));
```

Comment:
This can be derived from `postsByHour` as I mentioned in the HeatmapGrid component